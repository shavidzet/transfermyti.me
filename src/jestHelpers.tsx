/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
