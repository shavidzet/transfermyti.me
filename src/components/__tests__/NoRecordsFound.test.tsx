import React from "react";
import { render } from "~/jestHelpers";
import NoRecordsFound from "../NoRecordsFound";

describe("the <NoRecordsFound> component", () => {
  test("matches its snapshot with valid props", () => {
    const wrapper = render(<NoRecordsFound />);

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });
});
