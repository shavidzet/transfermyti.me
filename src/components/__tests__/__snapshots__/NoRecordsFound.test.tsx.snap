// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <NoRecordsFound> component matches its snapshot with valid props 1`] = `
.emotion-0 {
  font-size: 1.25rem;
  font-weight: 600;
  padding: 3rem 0;
  text-align: center;
}

<div
  class="emotion-0"
>
  No records found!
</div>
`;
